### ISSUE TITLE
Tests failing with `FileNotFoundException`

### ISSUE THREAD

**Issue by saurabharora90** [takahirom/roborazzi#377]
We've been using Roborazzi for a while. Recently we've seen these issues become more and more frequent on our CI. 

```
Execution failed for task ':features:myModule::finalizeTestRoborazziDebug'.
> java.io.FileNotFoundException: /home/runner/work/x-android/x-android/features/mymodule/build/test-results/roborazzi/results/1760836821606_com.myapp.DetailUiScreenshotTest.empty state.json (No such file or directory)
```

We need to retry a bunch of times before it goes through. We've updated to the latest 1.16.1 version and continue to face this issue. 

We have build cache setup on CI and even tried clearing the build cache but didn't have any luck with this.

**Comment by takahirom** [comment]
@saurabharora90
Thank you for reporting this. Could you provide the --info and --stacktrace logs for this issue?

**Comment by saurabharora90** [comment]
Can you provide an email address to share the logs?

**Comment by takahirom** [comment]
@SaurabhArora90 
Thank you for checking with the latest version. This is my email address: takam.dev@gmail.com

**Comment by saurabharora90** [comment]
Emailed the --info logs

**Comment by takahirom** [comment]
@saurabharora90 
Thank you. I was able to understand the situation from the log you provided. Could you also provide a --stacktrace log or point out where the crash occurred? I think it occurred here; is that correct?

[io.github.takahirom.roborazzi.RoborazziPlugin](https://github.com/takahirom/roborazzi/blob/main/include-build/roborazzi-gradle-plugin/src/main/java/io/github/takahirom/roborazzi/RoborazziPlugin.kt#L237)

```kotlin
              val results: List<CaptureResult> = resultDirFileTree.get().mapNotNull {
                if (it.name.endsWith(".json")) {
                  CaptureResult.fromJsonFile(it.path) // **here**
                } else {
                  null
                }
              }
```

**Comment by takahirom** [comment]
📝 

```kotlin
              val results: List<CaptureResult> = resultDirFileTree.get().mapNotNull {
                if (it.name.endsWith(".json")) {
                  CaptureResult.fromJsonFile(it.path) // **here**
                } else {
                  null
                }
              }
```

```kotlin
  const val resultDirPathFromBuildDir = "test-results/roborazzi/results/"
 
      val resultDirFileProperty =
        project.layout.buildDirectory.dir(RoborazziReportConst.resultDirPathFromBuildDir)
      val resultDirFileTree =
        resultDirFileProperty.map { it.asFileTree }
```

**Comment by takahirom** [comment]
@saurabharora90 
I believe your tests are running testReleaseUnitTest and testDebugUnitTest simultaneously. Is this intentional? You might resolve this by excluding testReleaseUnitTest using `-x testReleaseUnitTest`. Ideally, however, Roborazzi should create a separate directory for each variant.

**Comment by saurabharora90** [comment]
> @saurabharora90 
> I believe your tests are running testReleaseUnitTest and testDebugUnitTest simultaneously. Is this intentional? You might resolve this by excluding testReleaseUnitTest using `-x testReleaseUnitTest`. Ideally, however, Roborazzi should create a separate directory for each variant.

Yes we are running both variants. It is intentional for the time being. Though we plan to migrate away from that in the long run.

For stack trace, I'll try to get it you need them. Given the error doesn't happen all the time, it might take some time.

**Comment by saurabharora90** [comment]
We've disabled release for the past one week and have stopped seeing this issue

**Comment by takahirom** [comment]
Thanks. I tried to reproduce the problem in the integration tests but was unable to do so. Therefore, I'll wait for the person who has a project that can reproduce it.
https://github.com/takahirom/roborazzi/pull/380
